### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'filename'

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#214]
In GitLab by @wsanchezv on Dec 15, 2016, 10:35

*Please describe how you installed Flake8*

Into a venv:
```
build/bin/pip install --upgrade flake8
```

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
{
  "dependencies": [
    {
      "dependency": "setuptools",
      "version": "31.0.1"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.5.2",
    "system": "Darwin"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.5.3"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.2.0"
    },
    {
      "plugin": "pyflakes",
      "version": "1.3.0"
    }
  ],
  "version": "3.2.1"
}
```

*Please describe the problem or feature*

If you give flake8 a target that does not exist, it raises an exception:

```
wsanchez$ build/bin/flake8 xyzzy
Traceback (most recent call last):
  File "build/bin/flake8", line 11, in <module>
    sys.exit(main())
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/main/cli.py", line 16, in main
    app.run(argv)
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/main/application.py", line 322, in run
    self._run(argv)
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/main/application.py", line 306, in _run
    self.run_checks()
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/main/application.py", line 243, in run_checks
    self.file_checker_manager.start(files)
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/checker.py", line 371, in start
    self.make_checkers(paths)
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/checker.py", line 275, in make_checkers
    for argument in paths
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/checker.py", line 285, in <listcomp>
    if argument == filename or should_create_file_checker(filename)
  File "/Users/wsanchez/Dropbox/Developer/wsanchez/sample-klein-app/build/lib/python3.5/site-packages/flake8/checker.py", line 412, in __init__
    self.display_name = self.processor.filename
AttributeError: 'NoneType' object has no attribute 'filename'
```

A friendly error along the lines of "no such file or directory" would be better here.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 27, 2017, 13:27

I believe this is already fixed in master and will be released in 3.3.0.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 27, 2017, 13:27

closed
